gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -753,6 +777,16 @@ public Builder setAttemptDirectPathXds() {
       return this;
     }
 
+    /**
+     * Package-Private scope as it is used to test DirectPath functionality in tests. This overrides
+     * the computed systemProductName when the class is initialized.
+     */
+    @VisibleForTesting
+    Builder setSystemProductName(String systemProductName) {
+      this.systemProductName = systemProductName;
+      return this;
+    }
```

Comment:
Package-Private setter to allow tests to override the computed SystemProductName